File: src/jray/math/intersections/CubeSphere.java
package jray.math.intersections;

import jray.common.Vect3;

public class CubeSphere {

    public static boolean isSphereEnclosedByCube(Vect3 cCenter, double cWidthHalf, Vect3 sCenter, Double sRadius){
    	return Math.abs(cCenter.data[0]-sCenter.data[0])<cWidthHalf - sRadius&&
	           Math.abs(cCenter.data[1]-sCenter.data[1])<cWidthHalf - sRadius&&
	           Math.abs(cCenter.data[2]-sCenter.data[2])<cWidthHalf - sRadius;
    }
    
    public static boolean isSphereIntersectingCube(Vect3 cCenter, double cWidthHalf, Vect3 sCenter, Double sRadius){
    	return Math.abs(cCenter.data[0]-sCenter.data[0])<cWidthHalf + sRadius&&
	           Math.abs(cCenter.data[1]-sCenter.data[1])<cWidthHalf + sRadius&&
	           Math.abs(cCenter.data[2]-sCenter.data[2])<cWidthHalf + sRadius;
    }
}

File: JRayXLib/JRayXLib/Math/intersections/CubeSphere.cs
using JRayXLib.Shapes;

namespace JRayXLib.Math.intersections
{
    public class CubeSphere {

        public static bool IsSphereEnclosedByCube(Vect3 cCenter, double cWidthHalf, Vect3 sCenter, double sRadius)
        {
            return System.Math.Abs(cCenter.X - sCenter.X) < cWidthHalf - sRadius &&
                   System.Math.Abs(cCenter.Y - sCenter.Y) < cWidthHalf - sRadius &&
                   System.Math.Abs(cCenter.Z - sCenter.Z) < cWidthHalf - sRadius;
        }

        public static bool IsSphereIntersectingCube(Vect3 cCenter, double cWidthHalf, Vect3 sCenter, double sRadius)
        {
            return System.Math.Abs(cCenter.X - sCenter.X) < cWidthHalf + sRadius &&
                   System.Math.Abs(cCenter.Y - sCenter.Y) < cWidthHalf + sRadius &&
                   System.Math.Abs(cCenter.Z - sCenter.Z) < cWidthHalf + sRadius;
        }
    }
}
